tests/integration/components/sl-button-test.js

```diff
@@ -0,0 +1,206 @@
+        'button tag was rendered'
+    );
+});
+
+test( 'Default action is triggered when element is clicked event when clicked', function( assert ) {
+    assert.expect( 1 );
+
+    const externalAction = () => {
+        assert.ok(
+            true,
+            'External action was called'
+        );
+    };
+
+    this.on( 'externalAction', externalAction );
+
+    this.render( hbs`
+        {{#sl-button action="externalAction"}}
+            Default Text
+        {{/sl-button}}
+    ` );
+
+    this.$( '>:first-child' ).click();
+});
+
+test( 'Button is enabled by default', function( assert ) {
+    this.render( template );
+
+    assert.strictEqual(
+        this.$().is( ':disabled' ),
+        false,
+        'Component is disabled by default'
+    );
+});
+
+test( 'Button is disabled when disabled is set to true', function( assert ) {
+    this.render( hbs`
+        {{#sl-button disabled=true}}
+            Default Text
+        {{/sl-button}}
+    ` );
+
+    assert.strictEqual(
+        this.$( '>:first-child' ).is( ':disabled' ),
+        true,
+        'Component becomes disabled'
+    );
+});
+
+test( 'Expected default classes are applied', function( assert ) {
+    this.render( template );
+
+    assert.ok(
+        this.$( '>:first-child' ).hasClass( 'btn' ),
+        'Has class "btn"'
+    );
+
+    assert.ok(
+        this.$( '>:first-child' ).hasClass( 'sl-button' ),
+        'Has class "sl-button"'
+    );
+});
+
+test( 'Labels are correctly initialized', function( assert ) {
+    this.render( hbs`
+        {{#sl-button label="Test"}}
+            Default Text
+        {{/sl-button}}
+    ` );
+
+    assert.equal(
+        this.$( '>:first-child' ).text().trim(),
+        'Test',
+        'Expected label is present as text'
+    );
+});
+
+test( 'Correct size class is set', function( assert ) {
+    this.render( hbs`
+        {{#sl-button size="large"}}
+            Default Text
+        {{/sl-button}}
+    ` );
+
+    assert.ok(
+        this.$( '>:first-child' ).hasClass( 'btn-lg' ),
+        'Has expected class "btn-lg"'
+    );
+
+    this.render( hbs`
+        {{#sl-button size="small"}}
+            Default Text
+        {{/sl-button}}
+    ` );
+
+    assert.ok(
+        this.$( '>:first-child' ).hasClass( 'btn-sm' ),
+        'Has expected class "btn-sm"'
+    );
+
+    this.render( hbs`
+        {{#sl-button size="extra-small"}}
+            Default Text
+        {{/sl-button}}
+    ` );
+
+    assert.ok(
+        this.$( '>:first-child' ).hasClass( 'btn-xs' ),
+        'Has expected class "btn-xs"'
+    );
+});
+
+test( 'Theme class is set correctly', function( assert ) {
+    this.render( hbs`
+        {{#sl-button theme="success"}}
+            Default Text
+        {{/sl-button}}
+    ` );
+
+    assert.ok(
+        this.$( '>:first-child' ).hasClass( 'btn-success' ),
+        'Has expected class "btn-success"'
+    );
+});
+
+test( 'Label changes for pending state', function( assert ) {
```

Comment:
There is also a class applied to the component associated with these properties.  Likely in a separate test, but make sure this is asserted as well.